app/src/main/kotlin/com/wire/android/ui/home/gallery/MediaGalleryViewModel.kt

```diff
@@ -54,9 +69,54 @@ class MediaGalleryViewModel @Inject constructor(
         mediaGalleryViewState = mediaGalleryViewState.copy(screenTitle = conversationName)
     }
 
+    fun saveImageToExternalStorage() {
+        viewModelScope.launch {
+            withContext(dispatchers.io()) {
+                val imageData = getImageData(imageAssetId.conversationId, imageAssetId.messageId)
+                if (imageData is Success) {
+                    val defaultImageName = "Wire downloaded image ${getCurrentParsedDateTime()}.jpeg"
+                    fileManager.saveToExternalStorage(defaultImageName, imageData.decodedAsset) {
+                        onFileSavedToExternalStorage()
+                    }
+                } else {
+                    onSaveError()
```

Comment:
I think this error case it's not tested yet 👀 so many cats will be happy if we cover this too or later on as a TODO comment

Reply:
😹